src/runtime/program/test_program_execute.zig

```diff
@@ -0,0 +1,211 @@
+    allocator: std.mem.Allocator,
+    tc: *TransactionContext,
+    program: anytype,
+    instruction: anytype,
+    accounts_params: []const InstructionAccountInfoParams,
+) !InstructionContext {
+    const program_index = blk: {
+        for (tc.accounts, 0..) |account, index|
+            if (account.pubkey.equals(&program.ID))
+                break :blk index;
+        return error.CoulfNotFindProgramAccount;
+    };
+
+    var accounts = std.ArrayList(InstructionAccountInfo).init(allocator);
+    for (accounts_params) |account_params| {
+        if (account_params.index_in_transaction >= tc.accounts.len)
+            return error.AccountIndexOutOfBounds;
+        try accounts.append(.{
+            .pubkey = tc.accounts[account_params.index_in_transaction].pubkey,
+            .is_signer = account_params.is_signer,
+            .is_writable = account_params.is_writable,
+            .index_in_transaction = account_params.index_in_transaction,
+        });
+    }
+
+    return .{
+        .tc = tc,
+        .program_id = program.ID,
+        .program_index = @intCast(program_index),
+        .instruction = try bincode.writeAlloc(allocator, instruction, .{}),
+        .accounts = try accounts.toOwnedSlice(),
+    };
+}
+
+const InstructionAccountInfoParams = struct {
+    is_signer: bool = false,
+    is_writable: bool = false,
+    index_in_transaction: u16 = 0,
+};
+
+pub fn expectProgramExecuteResult(
+    allocator: std.mem.Allocator,
+    program: anytype,
+    instruction: anytype,
+    instruction_accounts_params: []const InstructionAccountInfoParams,
+    transaction_context_params: TransactionContextParams,
+    expected_transaction_context_params: TransactionContextParams,
+) !void {
+    var transaction_context = try createTransactionContext(
+        allocator,
+        transaction_context_params,
+    );
+    defer {
+        for (transaction_context.accounts) |account|
+            allocator.free(account.account.data);
+        allocator.free(transaction_context.accounts);
+    }
+
+    var instruction_context = try createInstructionContext(
+        allocator,
+        &transaction_context,
+        program,
+        instruction,
+        instruction_accounts_params,
+    );
+    defer {
+        allocator.free(instruction_context.instruction);
+        allocator.free(instruction_context.accounts);
+    }
+
+    try program.execute(allocator, &instruction_context);
+
+    const expected_transaction_context = try createTransactionContext(
+        allocator,
+        expected_transaction_context_params,
+    );
+    defer {
+        for (expected_transaction_context.accounts) |account|
+            allocator.free(account.account.data);
+        allocator.free(expected_transaction_context.accounts);
+    }
+
+    try expectTransactionContextEqual(expected_transaction_context, transaction_context);
+}
+
+pub fn expectTransactionAccountEqual(
+    expected: TransactionAccount,
+    actual: TransactionAccount,
```

Comment:
Minor feature request. Is it possible to pass the index through to this function so the index of account that failed the equal check can be logged? Ran into mismatch while setting up test for the vote program and it was not quickly obvious which of the accounts are not equal.